src/main/java/backend/model/user/DeletionServiceImpl.kt

```diff
@@ -0,0 +1,148 @@
+        sponsoringInvoiceRepository.save(invoice)
+    }
+
+    fun delete(location: Location) {
+        location.uploader = null
+        locationRepository.save(location)
+    }
+
+    fun delete(groupMessage: GroupMessage, user: User) {
+        groupMessage.messages
+                .filter { it.creator?.id == user.account.id }
+                .forEach {
+                    it.text = ""
+                    it.creator = null
+                }
+
+        groupMessage.users.removeIf { it.account.id == user.account.id }
+        messageRepository.save(groupMessage)
+    }
+
+    fun delete(sponsoring: Sponsoring) {
+        sponsoring.removeSponsor()
+        sponsoringRepository.save(sponsoring)
+    }
+
+    fun delete(team: Team, user: User) {
+        team.members.removeIf { it.account.id == user.account.id }
+        teamRepository.save(team)
+    }
+
+    fun delete(challenge: Challenge) {
+        challenge.removeSponsor()
+        challenge.description = ""
+        challengeRepository.save(challenge)
+    }
+
+    fun deleteComments(posting: Posting, user: User) {
+        posting.comments
+                .filter { it.user?.account?.id == user.account.id }
+                .forEach {
+                    it.user = null
+                    it.text = ""
+                }
+
+        postingRepository.save(posting)
+    }
+
+    fun delete(posting: Posting) {
+        if (posting.challenge == null && posting.comments.isEmpty()) {
+            postingRepository.delete(posting)
+        } else {
+            posting.media = null
+            posting.user = null
+            posting.text = ""
+            postingRepository.save(posting)
+        }
+    }
+
+    @Transactional
+    fun deleteUserGeneratedData(user: User) {
+        user.account.getRole(Sponsor::class)?.let {
+            sponsoringInvoiceRepository.findBySponsorId(it.id!!).forEach(::delete)
+        }
+
+        sponsoringRepository.findBySponsorAccountId(user.account.id!!).forEach(::delete)
+        challengeRepository.findBySponsorAccountId(user.account.id!!).forEach(::delete)
+
+        messageRepository
+                .findWhereUserHasSentMessages(user.account.id!!)
+                .forEach { delete(it, user) }
+
+        user.account.groupMessages.forEach { delete(it, user) }
+
+        user.account
+                .getRole(Participant::class)
+                ?.getAllTeams()
+                ?.forEach { delete(it, user) }
+
+        postingRepository
+                .findAllByUserId(user.account.id!!)
+                .forEach(::delete)
+
+        postingRepository
+                .findAllLikedByUser(user.account.id!!)
+                .forEach {
+                    it.unlike(user.account)
+                    postingRepository.save(it)
+                }
+
+        postingRepository
+                .findAllCommentedByUser(user.account.id!!)
+                .forEach { deleteComments(it, user) }
+
+        locationRepository
```

Comment:
this will change donation sums, I would not do this.

Reply:
No it won't change it. The location is still assigned to the team. It's just no longer assigned to the user

Reply:
I think I will rename them to `anonymize` to make it clear

Reply:
You can see in the picture here: https://github.com/BreakOutEvent/breakout-frontend/pull/367

That it didn't change the donation promise or the score of the team